packages/angular_devkit/build_angular/src/angular-cli-files/models/webpack-configs/typescript.ts

```diff
@@ -122,10 +122,17 @@ export function getAotConfig(wco: WebpackConfigOptions, i18nExtract = false) {
   }
 
   const test = /(?:\.ngfactory\.js|\.ngstyle\.js|\.tsx?)$/;
+  const optimize = wco.buildOptions.optimization.scripts;
 
   return {
     module: { rules: [{ test, use: loaders }] },
-    plugins: [_createAotPlugin(wco, { tsConfigPath }, i18nExtract)]
+    plugins: [
+      _createAotPlugin(
```

Comment:
I assume that this really affects only aot+optimize setups, because we don't want to strip this metadata for jit+optimize builds.

Reply:
This transformer only removes the calls for files emitted by the TS compilation. If it's just a JIT compilation, user files won't have them at all. 

This behaviour begs the question though: are these calls present in js files inside node modules? Should they be removed from there too? What happens to workspace libraries built with Ivy (even though they shouldn't be built with Ivy for production)?

Reply:
The shared application module technique is the prime benefactor of this change.  Since this is a common technique, the change here should provide a benefit for a large segment of users.  For application code, this is essentially identical to the current state of decorator removal in VE.

For libraries, these calls may be present depending on how the library is built. If present, they will still be stripped at the later stages when using the CLI via the existing methods.  Since libraries are generally bundled prior to packaging, this PR wouldn’t apply even if it did process them as there are no imports to remove.  Later stage DCE should be fairly effective in those cases though and is essentially equivalent to the current method of removing downlevel decorators from libraries in VE when using the CLI.